File: TESTDB/src/data/Discipline.java
package data;

import javax.persistence.*;
import java.util.HashMap;
import java.util.Map;

@Entity
public class Discipline {
 
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private boolean angry = false;
    @Transient
    private Map<String,Boolean> map= new HashMap<>();

    Discipline() {
    }

    private int getId() {
        return id;
    }

    private void setId(int id) {
        this.id = id;
    }

    private boolean isAngry() {
        return angry;
    }

    private void setAngry(boolean angry) {
        this.angry = angry;
    }

    Map<String, Boolean> getNeeds() {
        if (!isAngry()) {
            map.put("Börje is angry", isAngry());
        }
        return map;
    }
}

File: TESTDB/src/Data/Discipline.kt
package Data

import java.util.LinkedHashMap
import javax.persistence.*

@Entity
class Discipline internal constructor() : Care.CareInformation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0
    var angry: Boolean = false

    @Transient
    private val map = LinkedHashMap<String, Boolean>()

    override fun newUnicorn(): Discipline {
        map["angry"] = false
        return this
    }


      override fun categories(): LinkedHashMap<String, Boolean> {
        if (!angry) {
            map["angry"] = angry
        }
        return map
    }

    override fun save(key: String): String {
        if (key == "angry") {
            angry = true
            map.remove(key)
        }
        return "is no longer $key"
    }

    override fun toString(): String {
        return javaClass.simpleName
    }
}